docs/language_guide.md

```diff
@@ -0,0 +1,88 @@
+
+A second goal is that package relationships should be clear in Bosatsu. All external names must be explicitly imported, and all values are private unless exported. To see the dependencies between packages no type inference is required.
+
+## Problem domain for Bosatu
+In the current programming landscape, the vast majority of code is either in Turing complete languages or very anemic data-structure languages (JSON, YAML, TOML). Bosatsu is exploring an intermediate space. Bosatu is not designed as a general purpose language. Bosatsu in a total language: all functions return values. This rules out both infinite loops and non-explicit failure possibilities (exceptions, crashes, etc...). The goal of Bosatsu is somewhat similar to Google's [starlark](https://github.com/bazelbuild/starlark) language, a python like language for configuration and embedded applications. Starlark is used to configure bazel builds, but you can imagine configuring systems currently configured with yaml or json, or replacing general purpose languages in systems configured with those (e.g. Scala's sbt is configured with scala).
+
+## Differences with Starlark
+The biggest three differences between Bosatsu and Starlark is that Bosatsu is statically typed, all values are immutable, and there are no side-effects. The smaller differences come from adding features found in functional programming that Python lacks: a more powerful anonymous function syntax as well as the ability to define enums similar to Rust.
+
+# Language Features
+1. A static type system using a generalization of [Hindley-Milner (or Damas-Milner)](https://en.wikipedia.org/wiki/Hindley%E2%80%93Milner_type_system) published in [this paper by Simon Peyton Jones et. al.](https://www.microsoft.com/en-us/research/publication/practical-type-inference-for-arbitrary-rank-types/)
+1. data classes or named tuples, which we call structs following Rust
+1. sum types, which we call enums following Rust.
+1. a package system with explicit import and export syntax.
+1. a highly python-inspired syntax including function definition, literal integers and strings, literal lists, list comprehensions, tuples [(todo)](https://github.com/johnynek/bosatsu/issues/18) and (string key) dictionaries [(todo)](https://github.com/johnynek/bosatsu/issues/14). Generally, if a syntax is possible to support in the current type system with immutable values, we generally intend to support it.
+
+There are also some un-features, or items we don't currently and may never support
+1. There is no recursion at all. The name of the current function or let binding is not in scope as we are creating a value, so it is impossible to explicitly write recursion or infinite data structures. 
```

Comment:
You do say "explicitly" here but I'm curious, is it possible to implement a fixed-point combinator (like the Y combinator) in bosatsu and still end up with recursion?